### PR TITLE
Added msbuild build of project in buildtest action

### LINKED ISSUE #1009: [Test] Add unit test for source code  generator
Add unit test for source code generators

Ref #1008 

How reproduce:
- Clean solution
- Install nuget: Microsoft.CodeAnalysis.CSharp 4.3.0
- Run: msbuild .\src\KubernetesClient.Models\ -t:restore -p:RestorePackagesConfig=true
- Run: msbuild .\src\KubernetesClient.Models\

### PR DESCRIPTION
**Pull request by m3nax**
Close #1009

### CODE REVIEW

**Comment by tg123** [comment]
what is the output of `test broken version`
looks the same as `dotnet build` to me

**Comment by m3nax** [comment]
@tg123 I have retested the pipeline and it seems fine.

**Comment by tg123** [comment]
well this is really weird to me
thanks for finding it out

/LGTM

**Review by tg123 — APPROVED** [review]

File: .github/workflows/buildtest.yaml [hunk 1]
```diff
@@ -36,6 +36,26 @@ jobs:
       #     path: tests/KubernetesClient.Tests/coverage.netcoreapp2.1.cobertura.xml
       #     repo_token: ${{ secrets.GITHUB_TOKEN }}
       #     minimum_coverage: 0
+    
+  # Test code gen for visual studio compatibility >> https://github.com/kubernetes-client/csharp/pull/1008
+  codgen:
+    runs-on: windows-latest
+    name: MSBuild build
```
Comment (WeihanLi) [inline .github/workflows/buildtest.yaml:43]: Is there some wrong with the `dotnet build`

Comment (m3nax) [reply .github/workflows/buildtest.yaml:43]: Yes, there is a weird bug:
- with msbuild and the Microsoft.CodeAnalysis.CSharp Version = 4.3.0 package the project does not compile (https://github.com/kubernetes-client/csharp/actions/runs/3087969401/jobs/4993905209#step:6: 41) and prevented it from being used with Visual Studio.
- with dotnet build the project compiled correctly.

A few days ago the brach master had all actions completed successfully but did not compile when opened with visual studio.

If you want to check, set the Microsoft.CodeAnalysis.CSharp package version to 4.3.0, clean the project and compile it with visual studio.

**Pull request merged**